### PR TITLE
CSV reader and Data Container 

### PR DESCRIPTION
**Pull request by rialeksandrov**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -6,25 +6,31 @@ set(CMAKE_CXX_STANDARD 17)
...
+		gradient_boosting/binarization/BinCreator.h
+		gradient_boosting/binarization/BinCreatorByAbsoluteValue.h
+		gradient_boosting/binarization/BinCreatorByStatistics.h
+		gradient_boosting/config/GradientBoostingConfig.h
+		gradient_boosting/loss_functions/GradientBoostingLossFunction.h
+		gradient_boosting/loss_functions/GradientBoostingMSELossFunction.h
+		gradient_boosting/trees/GradientBoostingTree.h
+		gradient_boosting/trees/GradientBoostingTreeOblivious.h
+		utils/data_containers/ElementContainer.h
+        utils/data_containers/DataContainer.h
```
Comment (lakhtanov) [inline CMakeLists.txt]: something's wrong with indentation

File: utils/io/SimpleCSVReader.cpp [hunk 2]
```diff
@@ -0,0 +1,42 @@
+#include "utils/io/SimpleCSVReader.h"
+
+#include <vector>
+#include <string>
+#include <fstream>
+
+namespace utils {
+namespace io {
+
+// TODO(rialeksandrov) move to utils/algorithms ?
```
Comment (lakhtanov) [inline utils/io/SimpleCSVReader.cpp]: yep, probably it'd be a great idea to move it to utils

File: utils/io/SimpleCSVReader.cpp [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    }
+  }
+  return result;
+}
+
+SimpleCSVReader::SimpleCSVReader(const std::string &file_name)
+    : CSVReaderBase(file_name) {
+}
+
+data_containers::DataContainer SimpleCSVReader::ReadFile() const {
```
Comment (lakhtanov) [inline utils/io/SimpleCSVReader.cpp]: probably it'd be a good idea to make header (list of names of columns) to be an optional param

File: utils/io/SimpleCSVReader.cpp [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+#include <vector>
+#include <string>
+#include <fstream>
+
+namespace utils {
+namespace io {
+
+// TODO(rialeksandrov) move to utils/algorithms ?
+std::vector<std::string> split(const std::string &str, char token = ',') {
```
Comment (lakhtanov) [inline utils/io/SimpleCSVReader.cpp]: https://stackoverflow.com/questions/236129/the-most-elegant-way-to-iterate-the-words-of-a-string
here's quite a good implementation of split function

File: utils/data_containers/DataContainer.cpp [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+    , validated_(false) {
+  data_.resize(rows_);
+  for (size_t row_idx = 0; row_idx < table.size(); ++row_idx) {
+    const auto& row = table[row_idx];
+    assert(row.size() == columns_);
+    for (const auto& el : row) {
+      data_[row_idx].emplace_back(el);
+    }
+  }
+  Validation();
```
Comment (lakhtanov) [inline utils/data_containers/DataContainer.cpp]: I'm not sure if it's a good idea to make a complex validation in a constructor
any ideas on how to move it out of constructor?

File: utils/io/SimpleCSVReader.cpp (hunk 4 above)
Comment (rialeksandrov) [reply utils/io/SimpleCSVReader.cpp]: it is okay for now

**Pull request merged**